src/full-write.c

```diff
@@ -39,14 +40,36 @@ extern int errno;
    Return LEN upon success, write's (negative) error code otherwise.  */
 
 int
-full_write (int desc, const char *ptr, size_t len)
+full_write (int desc, const char *ptr, size_t len, int diffwr)
 {
   int total_written;
 
   total_written = 0;
   while (len > 0)
     {
-      int written = write (desc, ptr, len);
+      int written = 0;
+      if (diffwr) { /* Check destination block content is same as the buffer */
+        char *rptr = 0;
```

Comment:
Consider using `char *rptr = NULL;` instead to make it easier looking for pointer NULL assignment.